src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1250,6 +1250,9 @@ public function getFormat($mimeType)
             if (in_array($mimeType, (array) $mimeTypes)) {
                 return $format;
             }
+            if (isset($canonicalMimeType) && in_array($canonicalMimeType, (array) $mimeTypes)) {
```

Comment:
But this would force to call `in_array()` which is slower in case there is no cannonical mime-type, the `isset()` here prevents this useless check (`in_array(null, (array) $mimeTypes)`).

Instead of `isset()` when the variable is `null` could be strict check:
```php
if (null !== $canonicalMimeType && in_array($canonicalMimeType, (array) $mimeTypes)) {}